apps/examples/mediaplayer/mediaplayer_main.cpp

```diff
@@ -338,12 +382,104 @@ class MediaPlayerController
 };
 
 extern "C" {
+
+static bool sg_wifiIsConnected = false;
+static std::mutex sg_wifiMutex;
+static std::condition_variable sg_wifiCondv;
+static const std::chrono::seconds WAIT_WIFI_TIMEOUT(5);
+
+#define WIFI_CONNECTED_RESETFLAG() sg_wifiIsConnected = false
+
+#define WIFI_CONNECTED_CHECKFLAG() sg_wifiIsConnected
+
+#define WIFI_CONNECTED_NOTIFY() \
+	do {\
+		std::lock_guard<std::mutex> lock(sg_wifiMutex);\
+		sg_wifiIsConnected = true;\
+		sg_wifiCondv.notify_one();\
+	} while (0)
+
+#define WIFI_CONNECTED_WAIT() \
+	do {\
+		std::unique_lock<std::mutex> lock(sg_wifiMutex);\
+		if (!sg_wifiCondv.wait_for(lock, WAIT_WIFI_TIMEOUT, []{return sg_wifiIsConnected;})) {\
+			printf("wifi connecting timeout!\n");\
+		}\
+	} while (0)
+
+static void wifi_sta_connected(wifi_manager_result_e result) {
+	printf("[%s] result %d\n", __FUNCTION__, (int)result);
+	WIFI_CONNECTED_NOTIFY();
+}
+
+static void wifi_sta_disconnected(wifi_manager_disconnect_e result) {
+	printf("[%s] result %d\n", __FUNCTION__, (int)result);
+}
+
+static int wifi_connect()
+{
+	static wifi_manager_cb_s wifi_callbacks = {
+		wifi_sta_connected,
+		wifi_sta_disconnected,
+		NULL,
+		NULL,
+		NULL,
+	};
+
+	wifi_manager_result_e ret = wifi_manager_init(&wifi_callbacks);
+	if (ret != WIFI_MANAGER_SUCCESS) {
+		printf("wifi_manager_init failed, ret %d\n", (int)ret);
+		return -1;
+	}
+
+	wifi_manager_ap_config_s config;
+	config.ssid_length = strlen(CONFIG_EXAMPLES_MEDIAPLAYER_SSID);
+	config.passphrase_length = strlen(CONFIG_EXAMPLES_MEDIAPLAYER_PASSPHRASE);
+	strncpy(config.ssid, CONFIG_EXAMPLES_MEDIAPLAYER_SSID, config.ssid_length + 1);
+	strncpy(config.passphrase, CONFIG_EXAMPLES_MEDIAPLAYER_PASSPHRASE, config.passphrase_length + 1);
+	config.ap_auth_type = (wifi_manager_ap_auth_type_e)CONFIG_EXAMPLES_MEDIAPLAYER_AUTHENTICATION;
+	config.ap_crypto_type = (wifi_manager_ap_crypto_type_e)CONFIG_EXAMPLES_MEDIAPLAYER_CRYPTO;
+
+	WIFI_CONNECTED_RESETFLAG();
+
+	ret = wifi_manager_connect_ap(&config);
+	if (ret != WIFI_MANAGER_SUCCESS) {
+		printf("wifi_manager_connect_ap failed, ret %d\n", (int)ret);
+		return -1;
+	}
+
+	WIFI_CONNECTED_WAIT();
```

Comment:
You should return -1 in the timeout case.

Reply:
Fixed